### PR TITLE
feat: support dynamic enable of rust backtrace

### PR DESCRIPTION
**Pull request by dqhl76**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

feat: support dynamic enable of rust backtrace

```
SYSTEM ENABLE EXCEPTION_BACKTRACE;
SYSTEM DISABLE EXCEPTION_BACKTRACE;
```

close: #15468 

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15749)

### CODE REVIEW

**Comment by Xuanwo** [comment]
Can we reuse existing stetement like `set`? Adding a new set `enable`/`disable` seems not cool.

**Comment by sundy-li** [comment]
maybe 

```
system enable EXCEPTION_BACKTRACE
```

**Comment by BohuTANG** [comment]
Looks better to add a setting that are easily enabled on the Cloud, as I believe these settings are intended for us rather than for users.

File: src/query/service/src/interpreters/interpreter_set_backtrace.rs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    fn is_ddl(&self) -> bool {
+        false
+    }
+
+    #[async_backtrace::framed]
+    #[minitrace::trace]
+    async fn execute2(&self) -> Result<PipelineBuildResult> {
+        change_backtrace(self.plan.switch);
```
Comment (zhang2014) [inline src/query/service/src/interpreters/interpreter_set_backtrace.rs:48]: Need broadcast to other nodes of cluster.

**Comment by zhang2014** [comment]
> Looks better to add a setting that are easily enabled on the Cloud, as I believe these settings are intended for us rather than for users.

Setting always set to tenant. We need admin API.

**Comment by dqhl76** [comment]
Thanks for reviews. I will keep the syntax and fix the missing boardcast problem.

File: src/query/ast/src/parser/statement.rs [hunk 2]
```diff
@@ -2063,7 +2070,8 @@ pub fn statement_body(i: Input) -> IResult<Statement> {
             | #show_locks : "`SHOW LOCKS [IN ACCOUNT] [WHERE ...]`"
             | #kill_stmt : "`KILL (QUERY | CONNECTION) <object_id>`"
             | #vacuum_temp_files : "VACUUM TEMPORARY FILES [RETAIN number SECONDS|DAYS] [LIMIT number]"
-            | #set_priority: "SET PRIORITY (HIGH | MEDIUM | LOW) <object_id>"
+            | #set_priority: "`SET PRIORITY (HIGH | MEDIUM | LOW) <object_id>`"
+            | #set_backtrace: "`(ENABLE | DISABLE) EXCEPTION_BACKTRACE`"
```
Comment (Xuanwo) [inline src/query/ast/src/parser/statement.rs:2074]: I strongly oppose the idea of introducing two new keywords for this feature. Can we reuse an existing one like `SET`? For example:

```sql
SET EXCEPTION_BACKTRACE TRUE
SET EXCEPTION_BACKTRACE FALSE
```

**Review by Xuanwo — CHANGES_REQUESTED** [review]

File: src/query/ast/src/parser/statement.rs (hunk 2 above)
Comment (dqhl76) [reply src/query/ast/src/parser/statement.rs:2074]: I found @zhang2014 said that `SET` is used for tenant. I am not sure if it lead to misunderstanding.

Comment (dqhl76) [reply src/query/ast/src/parser/statement.rs:2074]: I checked our [document](https://docs.databend.com/sql/sql-commands/administration-cmds/set-global). Could we add `GLOBAL` parameter to `SET` command to express this 
semantic. e.g.`SET GLOBAL EXCEPTION_BACKTRACE=TRUE`

File: src/query/service/src/interpreters/interpreter_set_backtrace.rs (hunk 1 above)
Comment (dqhl76) [reply src/query/service/src/interpreters/interpreter_set_backtrace.rs:82]: solved

File: src/query/ast/src/parser/statement.rs (hunk 2 above)
Comment (zhang2014) [reply src/query/ast/src/parser/statement.rs:2074]: We can introduce the system statement.

Comment (Xuanwo) [reply src/query/ast/src/parser/statement.rs:2074]: > We can introduce the system statement(SYSTEM ENABLE BACKTRACE).

How will user use it in sql? Does it look like the following:

```sql
system enable EXCEPTION_BACKTRACE
```

Comment (Xuanwo) [reply src/query/ast/src/parser/statement.rs]: Do we really need `ENABLE` and `DISABLE`? Can we reuse `SET` so users can:

```rust
system set EXCEPTION_BACKTRACE=true
```

Please note `EXCEPTION_BACKTRACE` is not a keyward too, just like `SET max_threads = 4;`

Comment (dqhl76) [reply src/query/ast/src/parser/statement.rs]: I feel `system set EXCEPTION_BACKTRACE=true` is more reasonable. What do you think? @zhang2014

Comment (zhang2014) [reply src/query/ast/src/parser/statement.rs]: Enable disable will be more explicit.

File: src/query/ast/src/ast/statements/statement.rs [hunk 3]
```diff
@@ -317,6 +317,11 @@ pub enum Statement {
         priority: Priority,
         object_id: String,
     },
+
+    // Enable or disable Rust Backtrace
+    SetBacktrace {
```
[the remark this answers is not in the document]
Comment (dqhl76) [reply src/query/ast/src/ast/statements/statement.rs]: solved

File: src/query/service/src/interpreters/interpreter_factory.rs [hunk 4]
```diff
@@ -598,6 +599,10 @@ impl InterpreterFactory {
                 ctx,
                 *p.clone(),
             )?)),
+            Plan::SetBacktrace(p) => Ok(Arc::new(SetBacktraceInterpreter::try_create(
```
Comment (zhang2014) [inline src/query/service/src/interpreters/interpreter_factory.rs:602]:
```suggestion
            Plan::System(p) => Ok(Arc::new(SetBacktraceInterpreter::try_create(
```

File: src/common/tracing/src/panic_hook.rs [hunk 5]
```diff
@@ -32,11 +36,15 @@ pub fn set_panic_hook() {
 }
 
 pub fn log_panic(panic: &PanicInfo) {
-    let backtrace = Backtrace::force_capture();
-    let backtrace_str = format!("{:?}", backtrace);
+    let backtrace_str = if ENABLE_BACKTRACE.load(Ordering::Relaxed) {
+        let backtrace = Backtrace::force_capture();
```
Comment (sundy-li) [inline src/common/tracing/src/panic_hook.rs:40]: also need to respect this flag in `enable_rust_backtrace`

**Review by zhang2014 — APPROVED** [review]

**Review by Xuanwo — APPROVED** [review]
Thanks!

**Comment by wubx** [comment]
@soyeric128  need doc

**Pull request merged**